client/me/reauth-required/index.jsx

```diff
@@ -245,30 +275,33 @@ class ReauthRequired extends Component {
 		const hasSmsRecoveryNumber =
 			!! this.props?.twoStepAuthorization?.data?.two_step_sms_last_four?.length;
 
+		const currentAuthType = enhancedSecurity ? 'webauthn' : twoFactorAuthType;
+
 		return (
 			<Dialog
 				autoFocus={ false }
 				className="reauth-required__dialog"
 				isFullScreen={ false }
 				isVisible={ this.props?.twoStepAuthorization.isReauthRequired() }
 			>
-				{ isSecurityKeySupported && twoFactorAuthType === 'webauthn' ? (
+				{ isSecurityKeySupported && currentAuthType === 'webauthn' ? (
 					<SecurityKeyForm
 						loginUserWithSecurityKey={ this.loginUserWithSecurityKey }
 						onComplete={ this.refreshNonceOnFailure }
 					/>
 				) : (
-					this.renderVerificationForm()
+					this.renderVerificationForm( enhancedSecurity )
```

Comment:
If you agree on what I suggested above on retrieving this from props, let's remove this arg.

```suggestion
					this.renderVerificationForm()
```